### PR TITLE
trainspace run dynamo table

### LINKED ISSUE #995: Creation of DynamoDB table for trainspace run

### PR DESCRIPTION
**Pull request by karkir0003**
# Trainspace Run DynamoDB Table

Github Issue Number Here: #995 
**What user problem are we solving?**
Creation of a separate trainspace run DynamoDB table to track a user's trainspace run attempts. This table lets us track a user's trainspace run attempt and store the S3 links to the result files. 

**What solution does this PR provide?**
SST resource of the trainspace run DynamoDB table stored in an infrastructure as code framework

**Testing Methodology**
Ran a dev deploy and took a look at the created dynamodb table to see if the partition key and global secondary indices are correct. We will also need to actually write data to the table in order to test that the CRUD operations work with this new table given valid data.

**Any other considerations**
Decide on how we can configure a Time to live (TTL) setting for data retention policy purposes

### CODE REVIEW

File: serverless/stacks/AppStack.ts [hunk 1]
```diff
@@ -11,6 +11,26 @@ export function AppStack({ stack }: StackContext) {
...
+      partitionKey: "runId",
+    },
+    globalIndexes: {
+      "TrainspaceIndex": { partitionKey: "trainspaceId", sortKey: "timestamp"},
+      "UserIndex": {partitionKey: "userId"}
+    },
+  });
   const api = new Api(stack, "Api", {
     authorizers: {
       FirebaseAuthorizer: {
```
Comment (dwu359) [inline serverless/stacks/AppStack.ts:36]: Add trainspaceRunTable information (name, arn, etc) to stack.addOutputs to output that in the terminal

**Review by dwu359 — CHANGES_REQUESTED** [review]
Small change with outputting the table info, other than that LGTM

File: serverless/stacks/AppStack.ts (hunk 1 above)
Comment (karkir0003) [reply serverless/stacks/AppStack.ts:36]: ok. do we really need the ARN since we are creating the table from scratch?

Comment (dwu359) [reply serverless/stacks/AppStack.ts:36]: Prob not actually, ig just the name and anything else that would be useful for identifying and debugging the table on AWS

Comment (karkir0003) [reply serverless/stacks/AppStack.ts:36]: added it in @dwu359

**Review by dwu359 — APPROVED** [review]

**Pull request merged**